nullaway/build.gradle

```diff
@@ -60,10 +59,20 @@ dependencies {
     epJavac deps.build.checkerDataflow
 }
 
+// We include and shade the checker framework jars into the NullAway jar, as we may have custom
+// changes.
 shadowJar {
+    // set classifier to null since we want the artifact uploaded to Maven Central to be the
+    // shadow jar.  Without this, the shadow jar is built with a '-all' suffix in the name.
     classifier = null
+    relocate "org.checkerframework", "shadow.checkerframework"
 }
-build.dependsOn shadowJar
+// Since we set classifier to null above, both the normal jar artifact and the shadow jar have
+// the same name, which can cause races if we are not careful.  We force shadowJar to depend on
+// assemble, so we know that the shadow jar will always overwrite the normal one.  We also force
+// test to depend on shadowJar, to avoid races between running tests and building the shadow jar.
+shadowJar.dependsOn assemble
+test.dependsOn shadowJar
```

Comment:
@kageiit I added this since I assumed the tests run with the final jar in the classpath, not the directory containing the output classes.  Is this correct?